### ISSUE TITLE
GPU performance increase with xformers

### ISSUE THREAD

**Issue by AestheticMayhem** [oobabooga/text-generation-webui#695]
**GPU performance increase with xformers**

Is it possible to implement xformers for better performance as in stable diffusion, i only have a gtx 1060 with 6g vram and 16g ram, would be cool if it works, just a suggestion, i know nothing about coding

**thank you for the good work**

**Comment by AestheticMayhem** [comment]
> I have the exact same setup as your's. I use xformers and the medvram flag in stable diffusion and it helps me a lot. Would appreciate having it here too.

here is the link to xformers https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Xformers

There is a guide for installation that involves copying a repository into the main folder of the program. Additionally, a whl file must be copied into the same main folder and activated. I am currently attempting to follow this guide, and I believe it should work as transformers were designed primarily for NLP tasks. However, as a novice, I am unsure about where to place the files, and I would greatly appreciate any assistance.

**Comment by QuantumAlignmentLabOnline** [comment]
> > I have the exact same setup as your's. I use xformers and the medvram flag in stable diffusion and it helps me a lot. Would appreciate having it here too.
> 
> here is the link to xformers https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Xformers
> 
> There is a guide for installation that involves copying a repository into the main folder of the program. Additionally, a whl file must be copied into the same main folder and activated. I am currently attempting to follow this guide, and I believe it should work as xformers were designed primarily for NLP tasks. However, as a novice, I am unsure about where to place the files, and I would greatly appreciate any assistance.
> 
> I have asked GPT multiple questions, and it has consistently confirmed that it should be possible to utilize xformers if they are correctly loaded and have appropriate instructions in the main run file to enable their use.
> 
> I am attempting to locate the same bridge instruction in the stable diffusion code in order to include it in the .py file of Oobabooga. Based on my research, it appears that the instruction will resemble the transformer section and the tokenizer will load it for calculation onto the CUDA.
> 
> Establishing an official Oobabooga Discord channel would provide a beneficial platform for the community to collaborate and enhance the program's development, as it shows promise.

Any updates?

**Comment by knoopx** [comment]
https://github.com/huggingface/transformers/pull/22386

**Comment by AestheticMayhem** [comment]
> Any updates?

I attempted to duplicate the xformer dataset and copied it to both the main oobabooga folder and the modules folder for extension-related configuration loads. In addition, I installed the necessary dependencies for the xformer with PyTorch 2 acceleration and a CUDA toolkit GPU. I also added a command line to the webui.bat file to enforce loading when running the program. While it recognizes the xformer, it still generates an error. I believe that we need to include instructions in the main server exe .py file to establish a relationship and comprehension of the xformers' functions when running the program.

**Comment by AestheticMayhem** [comment]
> related [huggingface/transformers#22386](https://github.com/huggingface/transformers/pull/22386)




thanks for sharing, i will get thru it

**Comment by AestheticMayhem** [comment]
This is a promising start, as it appears to be functional and applicable. I came across a link in the thread you provided that demonstrates the use of xformers with a different dataset. The links are https://github.com/Bayes-Song/Open-Llama/blob/main/README_en.md | http://home.ustc.edu.cn/~sl9292/ | https://github.com/Bayes-Song/Open-Llama. This dataset was originally used with xformers for model training and the calculation works well for text-to-text interpretation for computation and generation. I will attempt to duplicate the repository on top of oobabooga, using the same setup they used for training but for generation purposes. In addition to using the --xformer command when launching the program, it is recommended to use the accelerate command to ensure that the torch/cuda link is taken into consideration. I am hoping to complete this task by tonight.

**Comment by AestheticMayhem** [comment]
> [<img alt="@AestheticMayhem" width="40" height="40" src="https://avatars.githubusercontent.com/u/111237581?s=80&amp;v=4">](/AestheticMayhem)
> 
> Write Preview
> Add heading textAdd bold text, <Ctrl+b>Add italic text, <Ctrl+i>
> Add a quote, <Ctrl+Shift+.>Add code, <Ctrl+e>Add a link, <Ctrl+k>
> Add a bulleted list, <Ctrl+Shift+8>Add a numbered list, <Ctrl+Shift+7>Add a task list, <Ctrl+Shift+l>
> Directly mention a user or team
> Reference an issue, pull request, or discussion
> Add saved reply
> Slash commands
> 
> 
> Attach files by dragging & dropping, selecting or pasting them.[ ](https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)Styling with Markdown is supported
>  Close issue
> Comment
> Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines).

**Comment by AestheticMayhem** [comment]
> > related [huggingface/transformers#22386](https://github.com/huggingface/transformers/pull/22386)
> 
> thanks for sharing, i will get thru it




This is a promising start, as it appears to be functional and applicable. I came across a link in the thread you provided that demonstrates the use of xformers with a different dataset. The links are https://github.com/Bayes-Song/Open-Llama/blob/main/README_en.md | http://home.ustc.edu.cn/~sl9292/ | https://github.com/Bayes-Song/Open-Llama. This dataset was originally used with xformers for model training and the calculation works well for text-to-text interpretation for computation and generation. I will attempt to duplicate the repository on top of oobabooga, using the same setup they used for training but for generation purposes. In addition to using the --xformer command when launching the program, it is recommended to use the accelerate command to ensure that the torch/cuda link is taken into consideration. I am hoping to complete this task by tonight.

@AestheticMayhem

 
Add heading textAdd bold text, <Ctrl+b>Add italic text, <Ctrl+i>
Add a quote, <Ctrl+Shift+.>Add code, <Ctrl+e>Add a link, <Ctrl+k>
Add a bulleted list, <Ctrl+Shift+8>Add a numbered list, <Ctrl+Shift+7>Add a task list, <Ctrl+Shift+l>
Directly mention a user or team
Reference an issue, pull request, or discussion
Add saved reply
Slash commands
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Styling with Markdown is supported
Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines).
Assignees
No one assigned
Labels
[enhancement](https://github.com/oobabooga/text-generation-webui/labels/enhancement)
New feature or request
Projects
None yet
Milestone
No milestone
Development
No branches or pull requests

Notifications
Customize
You’re receiving notifications because you authored the thread.
4 participants
